From patchwork Thu Aug  6 12:43:56 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Mohammed Gamal <mgamal@redhat.com>
X-Patchwork-Id: 11703665
Return-Path: <SRS0=LWg4=BQ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 5083514B7
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  6 Aug 2020 17:21:32 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id ABB8E23120
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  6 Aug 2020 17:21:32 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="DHn4MoUR"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727976AbgHFRVb (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 6 Aug 2020 13:21:31 -0400
Received: from us-smtp-delivery-1.mimecast.com ([207.211.31.120]:35807 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1730267AbgHFRQk (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 6 Aug 2020 13:16:40 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1596734194;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=XFwy7e2/BEqkhEQtkFfwSeARVG8Dp3lgZH+uIout38Q=;
        b=DHn4MoURhq49EqHO1QDNa2kONPcRKjKPddgg36qfaitIM2XSOH6pHuccR3ruFkahLTqZWC
        QTZSbBnlbKKcy0fZki0ZQJ7Bp/c0mUGvrpQzyyvhloOnkLhMMVzXXrGpboH2ACri1m3G8q
        uZuTN8aJx1T8dbDv7luf8OpDdu0xHrY=
Received: from mimecast-mx01.redhat.com (mimecast-mx01.redhat.com
 [209.132.183.4]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-507-an0cKzITPa-P4KJwKUGBlA-1; Thu, 06 Aug 2020 08:44:22 -0400
X-MC-Unique: an0cKzITPa-P4KJwKUGBlA-1
Received: from smtp.corp.redhat.com (int-mx06.intmail.prod.int.phx2.redhat.com
 [10.5.11.16])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mimecast-mx01.redhat.com (Postfix) with ESMTPS id 61CFBA4769;
        Thu,  6 Aug 2020 12:44:11 +0000 (UTC)
Received: from localhost.localdomain.com (ovpn-114-17.ams2.redhat.com
 [10.36.114.17])
        by smtp.corp.redhat.com (Postfix) with ESMTP id 16B5F5C1BD;
        Thu,  6 Aug 2020 12:44:08 +0000 (UTC)
From: Mohammed Gamal <mgamal@redhat.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com
Cc: Mohammed Gamal <m.gamal005@gmail.com>
Subject: [kvm-unit-tests PATCH 1/3] unittests.cfg: Increase timeout for access
 test
Date: Thu,  6 Aug 2020 14:43:56 +0200
Message-Id: <20200806124358.4928-2-mgamal@redhat.com>
In-Reply-To: <20200806124358.4928-1-mgamal@redhat.com>
References: <20200806124358.4928-1-mgamal@redhat.com>
MIME-Version: 1.0
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.16
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Mohammed Gamal <m.gamal005@gmail.com>

In case of npt=0 on kvm_amd, tests run slower and timeout,
so increase the timeout. The tests then do succeed.

Signed-off-by: Mohammed Gamal <m.gamal005@gmail.com>
---
 x86/unittests.cfg | 1 +
 1 file changed, 1 insertion(+)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 3a79151..4fa42fa 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -117,6 +117,7 @@ extra_params = -cpu qemu64,+x2apic,+tsc-deadline -append tscdeadline_immed
 file = access.flat
 arch = x86_64
 extra_params = -cpu host,host-phys-bits
+timeout = 180
 
 [smap]
 file = smap.flat

From patchwork Thu Aug  6 12:43:57 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Mohammed Gamal <mgamal@redhat.com>
X-Patchwork-Id: 11703657
Return-Path: <SRS0=LWg4=BQ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1826C722
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  6 Aug 2020 17:17:34 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 7A63223124
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  6 Aug 2020 17:17:34 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="NYGfsOPA"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1730101AbgHFRRb (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 6 Aug 2020 13:17:31 -0400
Received: from us-smtp-delivery-1.mimecast.com ([205.139.110.120]:31251 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1730440AbgHFRRG (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 6 Aug 2020 13:17:06 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1596734224;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=gT8t13wbhbYFLFrPxxkSEHSrNjvXeJVY5ONNGfCIIHQ=;
        b=NYGfsOPAE/3+G2UZuk384ULEZGjPI3zGaYxZjoBHY3X0smipYmTsBXmP0F/APez6t7XTRo
        JN+JUzwfpNSr7EzpYxTwDVSWrwZFwYRRwtJHtWtpRqG/891pSQg9DoNIM4rpYfa/9r4i/C
        oixxw1/vQK3t2TAReFcw7ekIAiF++Us=
Received: from mimecast-mx01.redhat.com (mimecast-mx01.redhat.com
 [209.132.183.4]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-351-Qpunr_n1O12gwAKZSOfrvw-1; Thu, 06 Aug 2020 08:44:31 -0400
X-MC-Unique: Qpunr_n1O12gwAKZSOfrvw-1
Received: from smtp.corp.redhat.com (int-mx06.intmail.prod.int.phx2.redhat.com
 [10.5.11.16])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mimecast-mx01.redhat.com (Postfix) with ESMTPS id F346C102C88B
        for <kvm@vger.kernel.org>; Thu,  6 Aug 2020 12:44:18 +0000 (UTC)
Received: from localhost.localdomain.com (ovpn-114-17.ams2.redhat.com
 [10.36.114.17])
        by smtp.corp.redhat.com (Postfix) with ESMTP id 348825F9DC;
        Thu,  6 Aug 2020 12:44:11 +0000 (UTC)
From: Mohammed Gamal <mgamal@redhat.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com
Cc: Mohammed Gamal <mgamal@redhat.com>
Subject: [kvm-unit-tests PATCH 2/3] x86/access: Skip running guest physical
 bits tests on AMD with NPT enabled
Date: Thu,  6 Aug 2020 14:43:57 +0200
Message-Id: <20200806124358.4928-3-mgamal@redhat.com>
In-Reply-To: <20200806124358.4928-1-mgamal@redhat.com>
References: <20200806124358.4928-1-mgamal@redhat.com>
MIME-Version: 1.0
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.16
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Since we can't handle NPT VM exits properly. We won't run
guest physical bits tests if NPT is enabled.

Signed-off-by: Mohammed Gamal <mgamal@redhat.com>
---
 x86/access.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/x86/access.c b/x86/access.c
index 7dc9eb6..90c5fe4 100644
--- a/x86/access.c
+++ b/x86/access.c
@@ -973,6 +973,14 @@ static int ac_test_run(void)
         invalid_mask |= AC_PTE_BIT36_MASK;
     }
 
+    if (this_cpu_has(X86_FEATURE_NPT)) {
+        printf("NPT enabled. Skipping physical bits tests\n");
+        invalid_mask |= AC_PDE_BIT51_MASK;
+        invalid_mask |= AC_PTE_BIT51_MASK;
+        invalid_mask |= AC_PDE_BIT36_MASK;
+        invalid_mask |= AC_PTE_BIT36_MASK;
+    }
+
     if (this_cpu_has(X86_FEATURE_PKU)) {
         set_cr4_pke(1);
         set_cr4_pke(0);

From patchwork Thu Aug  6 12:43:58 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Mohammed Gamal <mgamal@redhat.com>
X-Patchwork-Id: 11703663
Return-Path: <SRS0=LWg4=BQ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1D668138C
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  6 Aug 2020 17:21:27 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 784482311C
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu,  6 Aug 2020 17:21:27 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=redhat.com header.i=@redhat.com
 header.b="QEcI+cpP"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1730254AbgHFRVW (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 6 Aug 2020 13:21:22 -0400
Received: from us-smtp-delivery-1.mimecast.com ([207.211.31.120]:45879 "EHLO
        us-smtp-1.mimecast.com" rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org
        with ESMTP id S1730293AbgHFRQk (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 6 Aug 2020 13:16:40 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=redhat.com;
        s=mimecast20190719; t=1596734194;
        h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
         to:to:cc:cc:mime-version:mime-version:
         content-transfer-encoding:content-transfer-encoding:
         in-reply-to:in-reply-to:references:references;
        bh=aL/EuDNAcbdbzH+y5QWYpfnkZltNgWzwMN6+1Nt56y4=;
        b=QEcI+cpPoC9Wied2KZUPVNHW7W5wyIufrqLrzxctluUlRKre9kFh7fuxqCA+hA6B61nnLn
        VNYyMYFNLod3ub4jQbReZ9CKjon5h/163kbwa7GM7zZXjBb4EqNt1fHca6DWX5MstQBKHP
        gAzpePGeIBbGja07tz7CoijyrkUKNsY=
Received: from mimecast-mx01.redhat.com (mimecast-mx01.redhat.com
 [209.132.183.4]) (Using TLS) by relay.mimecast.com with ESMTP id
 us-mta-157-H9nv_85bOuKjUfqAG14j7Q-1; Thu, 06 Aug 2020 08:44:33 -0400
X-MC-Unique: H9nv_85bOuKjUfqAG14j7Q-1
Received: from smtp.corp.redhat.com (int-mx06.intmail.prod.int.phx2.redhat.com
 [10.5.11.16])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mimecast-mx01.redhat.com (Postfix) with ESMTPS id B6D72803022;
        Thu,  6 Aug 2020 12:44:24 +0000 (UTC)
Received: from localhost.localdomain.com (ovpn-114-17.ams2.redhat.com
 [10.36.114.17])
        by smtp.corp.redhat.com (Postfix) with ESMTP id AF3E65F9DC;
        Thu,  6 Aug 2020 12:44:19 +0000 (UTC)
From: Mohammed Gamal <mgamal@redhat.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com
Cc: Mohammed Gamal <m.gamal005@gmail.com>
Subject: [kvm-unit-tests PATCH 3/3] Revert "access: disable phys-bits=36 for
 now"
Date: Thu,  6 Aug 2020 14:43:58 +0200
Message-Id: <20200806124358.4928-4-mgamal@redhat.com>
In-Reply-To: <20200806124358.4928-1-mgamal@redhat.com>
References: <20200806124358.4928-1-mgamal@redhat.com>
MIME-Version: 1.0
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.16
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

From: Mohammed Gamal <m.gamal005@gmail.com>

This reverts commit 71de9c93fbdb15978ffa2f290dd120d3005a9292.

Now that guest_maxphyaddr < host_maxphyaddr patches are sent.
Revert this change in kvm-unit-test

Signed-off-by: Mohammed Gamal <m.gamal005@gmail.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 4fa42fa..3c0b992 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -116,7 +116,7 @@ extra_params = -cpu qemu64,+x2apic,+tsc-deadline -append tscdeadline_immed
 [access]
 file = access.flat
 arch = x86_64
-extra_params = -cpu host,host-phys-bits
+extra_params = -cpu host,phys-bits=36
 timeout = 180
 
 [smap]
